### ISSUE TITLE
Session Replay not working, PixelCopy, Window doesn't have a backing

### ISSUE THREAD

**Issue by varunraghu3112** [getsentry/sentry-react-native#4147]
**OS:**
- [ ]  Windows
- [x]  MacOS
- [ ]  Linux

**Platform:**
- [ ]  iOS
- [x]  Android

**SDK:**
- [x]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:** 0.0.0

**`react-native` version:** 5.31.1

**Are you using Expo?**
- [ ]  Yes
- [x]  No

**Are you using sentry.io or on-premise?**
- [x]  sentry.io (SaaS)
- [ ]  on-premise

**If you are using sentry.io, please post a link to your issue so we can take a look:**

[[Link to issue](https://galactus-funware-technology-pr.sentry.io/issues/5946528120/replays/?project=1802014&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&statsPeriod=1h&stream_index=0&utc=true)]

**Configuration:**

(`@sentry/react-native`)
```
Sentry.init({
  dsn: NativeManager.sentry_dsn,
  release: `xxxx`,
  _experiments: {
    replaysSessionSampleRate: 1.0,
    replaysOnErrorSampleRate: 1.0
  },
  dist: '1000494',
  enableAutoSessionTracking: true,
  maxBreadcrumbs: 50,
  environment: __DEV__ ? 'development' : logEnabled ? 'qa' : 'production',
  ignoreErrors: [],
  tracesSampleRate: 1.0,
  sampleRate: 1.0,
  integrations: [
    Sentry.mobileReplayIntegration(),
    new Sentry.ReactNativeTracing({
      tracingOrigins: [NativeManager.baseUrl],
      routingInstrumentation
    })
  ]
});
```

---
**I have the following issue:**

I m trying to integrate Replays in to my project but its not working, on adb logs I can see Sentry only for three times after app launch and nothing about replays (Refer pics), I manually threw a JS exception to test the Replay feature. The errors are caught correctly by Sentry but no Replays there

**Steps to reproduce:**
- enable Session Replay from Sentry.init()
- throw a JS exception from the Android build
- find and view the exception in Sentry issues

**Actual result:**

No Session Replay is found for the exception![Image](https://github.com/user-attachments/assets/4939693a-a9df-46c8-a679-b02201b8d8ed)
![Image](https://github.com/user-attachments/assets/c6f85fa3-27f9-4498-80fb-7425a3452d85)

**Expected result:**

Session found for the exception

**Comment by varunraghu3112** [comment]
Hi @rwachtler I have enabled debug:true, I can also see  Capturing session replay: xxxx in logcat but in sentry Replays are still 0 (You can check on the above link)

![Image](https://github.com/user-attachments/assets/ab7b1a30-2658-4988-8879-687e7642ec9e)

**Comment by varunraghu3112** [comment]
Im just trying to throw error manually by "throw new error"![Image](https://github.com/user-attachments/assets/37190303-a79f-4cf3-ac93-c346e90345ca)
Errors are getting captured in Sentry but replays are 0

**Comment by varunraghu3112** [comment]
@rwachtler https://drive.google.com/file/d/1IDhsxnbhPq0wucobbIf00GLfyHRDOTtt/view?usp=sharing you can refer the filtered output of Sentry logs from logcat

**Comment by varunraghu3112** [comment]
![Image](https://github.com/user-attachments/assets/c2ea320f-b670-4cf0-b7ab-0a74bb60e4be)
Still in stats nothing has been hit here (For your reference)

**Comment by rwachtler** [comment]
@varunraghu3112 Thanks for providing the whole log! There are following things standing out to me:


- `Failed to determine view hierarchy, not capturing`
  - appears shortly after `Capturing session replay` - `Serializing object` / or slightly before `Envelope sent successfully`
  - see [sentry-java/ScreenRecorder.kt](https://github.com/getsentry/sentry-java/blob/b5b093e5f805d0d02c7be344403804e7a7605398/sentry-android-replay/src/main/java/io/sentry/android/replay/ScreenshotRecorder.kt#L119)
  - then usually the envelope is sent `Envelope sent successfully.` 
- There is a 
```bash
10-04 11:03:40.439  7969  7969 W Sentry  : Failed to capture replay recording
10-04 11:03:40.439  7969  7969 W Sentry  : java.lang.IllegalArgumentException: Window doesn't have a backing surface!
``` 

in the end. Which is coming from the [Screenrecorder.kt](https://github.com/getsentry/sentry-java/blob/b5b093e5f805d0d02c7be344403804e7a7605398/sentry-android-replay/src/main/java/io/sentry/android/replay/ScreenshotRecorder.kt#L113) as well

There are also some occurrences of:

```
SentryExecutorS: type=1400 audit(0.0:2090129): avc:  denied  { read } for  name="u:object_r:vendor_display_prop:s0" dev="tmpfs" ino=15603 scontext=u:r:untrusted_app:s0:c28,c261,c512,c768 tcontext=u:object_r:vendor_display_prop:s0 tclass=file permissive=0 app=com.mpl.androidapp
```

Maybe someone from the core team has an idea what might be the root-cause for this 🤔

**Comment by krystofwoldrich** [comment]
Thank you @varunraghu3112 and @rwachtler for gathering all the details.

I can confirm that the Replays are not recorded due to the `java.lang.IllegalArgumentException: Window doesn't have a backing surface!` exception, but at the moment I don't see why the [PixelCopy API](https://android.googlesource.com/platform/frameworks/base.git/+/master/graphics/java/android/view/PixelCopy.java#297) throws the error.

---

Does the error happen both on Emulator and also on your Device? 

Do you initialize Sentry Android manually or the only init is the `Sentry.init` call in React Native?

**Comment by krystofwoldrich** [comment]
@romtsn Can you check this out, looks like the [PixelCopy API](https://android.googlesource.com/platform/frameworks/base.git/+/master/graphics/java/android/view/PixelCopy.java#297) errors out.

**Comment by varunraghu3112** [comment]
> Thank you [@varunraghu3112](https://github.com/varunraghu3112) and [@rwachtler](https://github.com/rwachtler) for gathering all the details.
> 
> I can confirm that the Replays are not recorded due to the `java.lang.IllegalArgumentException: Window doesn't have a backing surface!` exception, but at the moment I don't see why the [PixelCopy API](https://android.googlesource.com/platform/frameworks/base.git/+/master/graphics/java/android/view/PixelCopy.java#297) throws the error.
> 
> Does the error happen both on Emulator and also on your Device?
> 
> Do you initialize Sentry Android manually or the only init is the `Sentry.init` call in React Native?

@krystofwoldrich 
- I haven't tried this on Emulator
- Sentry.init happens on React native side

**Comment by romtsn** [comment]
@varunraghu3112 could you give a little bit more details into:
- What device are you using for testing
- What is your app like? Do you have any animations in the app or perhaps a video view?
- Could you post a link to your project/org in sentry?

**Comment by krystofwoldrich** [comment]
[Link to Sentry](https://galactus-funware-technology-pr.sentry.io/issues/5946528120/replays/?project=1802014&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&statsPeriod=1h&stream_index=0&utc=true) from the issue desc.

**Comment by romtsn** [comment]
@varunraghu3112 I've figured out what the problem is - your org is a sales-led organization (meaning you have a dedicated customer manager leading your account), therefore we have disabled replay ingestion for your org, because we decided to start the beta rollout only for self-serve(=smaller) customers.

However, we're going to open up beta for everyone already next week, so stay tuned! I think your Sentry-dedicated person will reach out with details shortly after that.
